src/ray/common/task/task_spec.cc

```diff
@@ -3,6 +3,9 @@
 #include <sstream>
 
 #include "ray/util/logging.h"
+// the below line gives "src/ray/common/task/task_spec.cc:6:10:
+// fatal error: 'ray/raylet/worker_pool.h' file not found"
+// #include "ray/raylet/worker_pool.h"
```

Comment:
I think I'm getting something very basic wrong here, but I'm not sure what it is.

Reply:
Same error if I include it in `task_spec.h` instead: 

```
In file included from src/ray/raylet_client/raylet_client.cc:15:
In file included from bazel-out/darwin-opt/bin/_virtual_includes/raylet_client_lib/ray/raylet_client/raylet_client.h:26:
bazel-out/darwin-opt/bin/_virtual_includes/ray_common/ray/common/task/task_spec.h:14:10: fatal error: 'ray/raylet/worker_pool.h' file not found
#include "ray/raylet/worker_pool.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

Reply:
Could this just be a big circular import problem?

Reply:
Ah yeah I think this is a circular import. Because worker_pool already depends on task_spec, we need to instead move the `WorkerCacheKey` here and include it from here in worker_pool

Reply:
Nice thanks, made the change and it compiles now.